models/organization/team_invite.go

```diff
@@ -67,6 +113,11 @@ func CreateTeamInvite(ctx context.Context, doer *user_model.User, team *Team, em
 		}
 	}
 
+	// check if the user is already a team member by email
+	if err = isTeamMemberByEmail(ctx, team, email); err != nil {
+		return nil, err
+	}
+
```

Comment:
```suggestion
	exist, err := db.GetEngine(ctx).
		Where(builder.Eq{
			"team_user.org_id":  team.OrgID,
			"team_user.team_id": team.ID,
			"user.email":        email,
		}).
		Join("INNER", "user", "user.id = team_user.uid").
		Table("team_user").
		Exist()
	if err != nil {
		return nil, err
	}
	if exist {
		return nil, ErrUserEmailAlreadyAdded{
			Email: email,
		}
	}

```
How about a single query? Con is there is no username available anymore. But we don't show that info anyway.